.github/workflows/devnet-release.yml

```diff
@@ -3,56 +3,56 @@ name: "Devnet Community Release"
 on:
   push:
     branches:
-      - main
+      - devnet-image
 
 jobs:
   deploy-devnet:
-    runs-on: ubuntu-latest
+    runs-on:
+      - self-hosted
+      - linux
+      - x64
+      - sre
     concurrency:
       group: deploy-devnet
       cancel-in-progress: false
+    container:
+      image: niteo/nixpkgs-nixos-21.11:ed02c2ba0384b2800db41333045a6fb781f12aac
     steps:
     - uses: actions/checkout@v2
 
-    - uses: google-github-actions/setup-gcloud@master
-      with:
-        service_account_key: ${{ secrets.GCP_CREDENTIALS }}
-        export_default_credentials: true
-
-    - uses: actions/setup-python@v2
-
-    - uses: cachix/install-nix-action@v16
+    - name: Install Dependencies
+      run: |
+        nix-channel --add https://nixos.org/channels/nixpkgs-unstable nixpkgs
+        nix-channel --update
+        nix-env -iA nixpkgs.python3 nixpkgs.rustup nixpkgs.sccache nixpkgs.jq nixpkgs.google-cloud-sdk nixpkgs.nix nixpkgs.neofetch
+        nix --version
+        rustup install stable
```

Comment:
there are nix packages supporting stable and nightly as it seems. why not nix? ok. just may be leave comment why not.

Reply:
Because I don't want to pin stable